hoodie-client/src/main/java/com/uber/hoodie/io/storage/HoodieOrcWriter.java

```diff
@@ -0,0 +1,192 @@
+  private final Writer writer;
+  private final TypeDescription orcSchema;
+  private final List<String> orcFieldNames;
+  private final VectorizedRowBatch orcBatch;
+
+  public HoodieOrcWriter(String commitTime, Path file, HoodieOrcConfig orcConfig,
+      TypeDescription orcSchema) throws IOException {
+    this.file = HoodieWrapperFileSystem.convertToHoodiePath(file, orcConfig.getHadoopConf());
+    this.fs = (HoodieWrapperFileSystem) this.file.getFileSystem(registerFileSystem(file, orcConfig.getHadoopConf()));
+    this.writeSupport = orcConfig.getWriteSupport();
+    this.commitTime = commitTime;
+    this.stripeSize = orcConfig.getStripeSize();
+
+    this.orcSchema = orcSchema;
+    // hoodie meta field
+    this.orcSchema.addField("_hoodie_commit_time", TypeDescription.createString())
+        .addField("_hoodie_commit_seqno", TypeDescription.createString())
+        .addField("_hoodie_record_key", TypeDescription.createString())
+        .addField("_hoodie_partition_path", TypeDescription.createString())
+        .addField("_hoodie_file_name", TypeDescription.createString());
+
+    this.writer = OrcFile.createWriter(
+        file,
+        OrcFile.writerOptions(orcConfig.getHadoopConf())
+            .setSchema(this.orcSchema)
+            .stripeSize(orcConfig.getStripeSize())
+            .bufferSize(orcConfig.getBufferSize())
+            .blockSize(orcConfig.getBlockSize())
+            .compress(CompressionKind.ZLIB)
+            .version(org.apache.orc.OrcFile.Version.V_0_12));
+
+    orcFieldNames = this.orcSchema.getFieldNames();
+    orcBatch = this.orcSchema.createRowBatch();
+  }
+
+  public static Configuration registerFileSystem(Path file, Configuration conf) {
+    Configuration returnConf = new Configuration(conf);
+    String scheme = FSUtils.getFs(file.toString(), conf).getScheme();
+    returnConf.set("fs." + HoodieWrapperFileSystem.getHoodieScheme(scheme) + ".impl",
+        HoodieWrapperFileSystem.class.getName());
+
+    return returnConf;
+  }
+
+  @Override
+  public void writeAvroWithMetadata(R avroRecord, HoodieRecord record) throws IOException {
+    GenericRecord genericRecord = (GenericRecord) avroRecord;
+    String seqId = HoodieRecord.generateSequenceId(commitTime, TaskContext.getPartitionId(),
+        recordIndex.getAndIncrement());
+
+    HoodieAvroUtils.addHoodieKeyToRecord(genericRecord, record.getRecordKey(),
+        record.getPartitionPath(), file.getName());
+
+    GenericRecord metadataToRecord = HoodieAvroUtils
+        .addCommitMetadataToRecord(genericRecord, commitTime, seqId);
+
+    Schema avroSchema = metadataToRecord.getSchema();
+
+    int row = orcBatch.size++;
+    setOrcFiledValue((R) avroRecord, avroSchema, row);
+
+    if (orcBatch.size !=  0) {
+      writer.addRowBatch(orcBatch);
+      orcBatch.reset();
+    }
+    writeSupport.add(record.getRecordKey());
+  }
+
+  @Override
+  public boolean canWrite() {
+    return fs.getBytesWritten(file) < stripeSize;
```

Comment:
Is stripeSize the same as fileSize ?

Reply:
> Is stripeSize the same as fileSize ?

The stripe size for orc file. The writer stores the contents of the stripe in memory until this memory limit is reached and the stripe is flushed to the HDFS file and the next stripe started.